scripts/asm-installer/install_asm

```diff
@@ -1411,9 +1424,25 @@ is_workload_identity_enabled() {
   if [[ "${ENABLED}" = 'null' ]]; then false; fi
 }
 
+get_hub_workload_identity() {
+  local IDENTITY_PROVIDER
+  local IDENTITY
+  IDENTITY_PROVIDER="$(kubectl get memberships membership -o=json | jq .spec.identity_provider)"
+  IDENTITY="$(echo "${IDENTITY_PROVIDER}" | sed 's/^\"https:\/\/gkehub.googleapis.com\/projects\/\(.*\)\/locations\/global\/memberships\/\(.*\)\"$/\1 \2/g')"
+  read -r ENVIRON_PROJECT_ID HUB_MEMBERSHIP_ID <<EOF
+${IDENTITY}
+EOF
+  MESHAPI_POST_DATA='{"workloadIdentityPools":["'${ENVIRON_PROJECT_ID}'.hub.id.goog"]}'
+  info "Cluster is in the Hub of project ${ENVIRON_PROJECT_ID} with Membership ID ${HUB_MEMBERSHIP_ID}"
+}
+
 enable_workload_identity(){
+  # Enable workload identity is not needed for Hub IDNS as the cluster needs to be registered with Hub IDNS before the installation script
+  if [[ "${CA}" == "mesh_ca" && "${ENABLE_HUB_IDNS}" -eq 1 ]]; then
+    get_hub_workload_identity
+    return
+  fi
```

Comment:
Instead of putting logic here, how about in init_meshconfig:

```
init_meshconfig() {
get_token
if [[ use_hub_idns -eq 1 ]]; then
  POST_DATA="$(get_hub_workload_identity)"
  curl --post POST_DATA
else
  curl --post ''
fi
```

And then add an `echo "${MESHAPI_POST_DATA}"` at the end of `get_hub_workload_identity`. This would also let us get rid of the global MESHAPI_POST_DATA.